src/realm/object-store/sync/async_open_task.cpp

```diff
@@ -61,8 +61,7 @@ void AsyncOpenTask::start(AsyncOpenCallback async_open_complete)
         }
 
         auto config = coordinator->get_config();
-        if (config.sync_config && config.sync_config->flx_sync_requested &&
-            config.sync_config->subscription_initializer) {
+        if (config.sync_config && config.sync_config->flx_sync_requested) {
```

Comment:
Inside` RealmCoordinator::get_realm() `we do this before to run the callback:

```
#ifdef REALM_ENABLE_SYNC
    if (subscription_function) {
      ..... //logic
      subscription_function(....);
```

So it will never run if set to null or not set.

Reply:
True, but we may wait in the task if the user commited some subscriptions themselves (not through the callback).

Reply:
Yes, definitely… especially with the check based on the status of the latest active subscription. I restored the previous check... should be fine now.